src/monodroid/CMakeLists.txt

```diff
@@ -492,6 +492,7 @@ if(ANDROID)
       ${BIONIC_SOURCES_DIR}/cxa_guard.cc
       ${SOURCES_DIR}/cxx-abi/string.cc
       ${SOURCES_DIR}/cxx-abi/terminate.cc
+      ${SOURCES_DIR}/cxx-abi/vector.cc
```

Comment:
On a lark, I wanted to re-examine which public symbols we're exporting:

```
% $HOME/android-toolchain/ndk//toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-nm -D -C bin/Debug/lib/xamarin.android/xbuild/Xamarin/Android/lib/x86_64/libmono-android.debug.so
00000000000216b0 T JNI_OnLoad
0000000000021730 T Java_mono_android_Runtime_init
00000000000236e0 T Java_mono_android_Runtime_initInternal
0000000000029410 T Java_mono_android_Runtime_notifyTimeZoneChanged
0000000000023d70 T Java_mono_android_Runtime_propagateUncaughtException
0000000000023730 T Java_mono_android_Runtime_register
```

Good and understandable.

```
000000000003e480 T std::terminate()
0000000000014f80 T operator delete[](void*)
0000000000014f90 T operator delete[](void*, std::nothrow_t const&)
0000000000014fa0 T operator delete[](void*, unsigned long)
0000000000014f50 T operator delete(void*)
0000000000014f60 T operator delete(void*, std::nothrow_t const&)
0000000000014f70 T operator delete(void*, unsigned long)
0000000000014ef0 T operator new[](unsigned long)
0000000000014f30 T operator new[](unsigned long, std::nothrow_t const&)
0000000000014e90 T operator new(unsigned long)
0000000000014ed0 T operator new(unsigned long, std::nothrow_t const&)
```

Uh... is this still sensible or desirable?  `libmonodroid.so` is loaded by *Android*, so *we* don't control if it's loaded with `RTLD_GLOBAL` or `RTLD_LOCAL` set, and if `RTLD_GLOBAL` *is* used, I don't know what happens if a *real* C++ lib is also loaded in the process.

Is there a way to make these *weak* exports?

Reply:
They shouldn't be visible, and they weren't previously. Apparently the switch to the newer NDK with the newer libc++ changed the visibility at some point. I am able to hide most of them instead of:

```
0000000000011d04 T operator delete[](void*)
0000000000011ce0 T operator delete(void*)
0000000000011c9c T operator new[](unsigned long)
0000000000011c58 T operator new(unsigned long)
```

They are declared by the compiler in each compilation unit, without having to include `<new>`, I'll check tomorrow if there's a way to change their visibility.

Reply:
Makes me wonder if we need a "test" to verify which symbols are exported, so that we don't accidentally export symbols we don't want exported.

Reply:
I made the for `delete` and `new` operators weak symbols, I can't figure out how to hide them right now (can't find the built-in declaration of the functions in the compiler)